src/MSALWrapper/AuthFlow/AuthFlowExecutor.cs

```diff
@@ -69,5 +76,11 @@ public async Task<AuthFlowResult> GetTokenAsync()
 
             return result;
         }
+
+        private void SendTelemetryEvent(AuthFlowResult attempt, string authFlowName)
+        {
+            // will be implemented in the next PR
+            return;
+        }

```

Comment:
I'm not sure we need the helper method here yet. I'd rather see us collect the metrics about this auth flow and send it first before pre-abstracting. 

I think in this PR, it's fine to go ahead and actually send the event and will make the whole change easier to reason about seeing the full change in the same diff. 

To send an event, you need an event name, and an instance of `EventData`. Consider creating your EventData instance at the start of the for loop - one for each authFlow attempt. You can then add values to that eventData, and at the end of the loop, send the event constructing an event name based off the authFlowName.

Reply:
I thought of creating an instance of `EventData` inside each AuthFlow and then populating it inside the AuthFlow. When returning from any AuthFlow, this `EventData` instance will be returned along with the `AuthFlowResult`  like
```
return new AuthFlowResult(tokenResult, eventdata);
```

I will add all the changes inside one PR, I think that will make it easy to review.

Reply:
@kyle-rader add changes here. Let me know if this approach doesn't seem good.

Reply:
Generally I want to avoid having the auth logic itself know anything about Telemetry. I am _okay_ having the AuthFlowExector know about Telemetry given it's a meta class design to run multiple auth flows. But each auth flow I think should only return it's information, and the consumer can then decide what to send telemetry wise. The limits the responisbiltiy of every auth flow class, and forces knowledge about telemetry into 1 place. It also helps keep the telemetry schema per auth flow similar / the same.